README.md

```diff
@@ -1,184 +1,68 @@
-
-## Not using PoW for registrations like rendezvous
-
-That's why it takes forever to be developed ;)
+> `libp2p-stardust` is one of the multiple transports available for libp2p. `libp2p-stardust` incorporates both a transport and a discovery service that is facilitated by the stardust server, also available in this repo.
 
 ## Lead Maintainer
 
 [Maciej Krüger](https://github.com/mkg20001)
 
-## Description
+## Protocol
 
-`libp2p-stardust` is one of the multiple transports available for libp2p. `libp2p-stardust` incorporates both a transport and a discovery service that is facilitated by the stardust server, also available in this repo and module.
+You can read the protocol specification on [PROTOCOL.md](./PROTOCOL.md).
 
-## Todo
+## Usage
 
- - [ ] Integrate as libp2p transport (WIP)
- - [x] Add discovery
- - [ ] Split server into it's own repo
+### Install
 
-## Usage
+```bash
+> npm install libp2p-stardust
+```
 
 ### Example
 
 ```js
-'use strict'
-
-const Libp2p = require('libp2p')
-const Id = require('peer-id')
-const Info = require('peer-info')
+const Stardust = require('libp2p-stardust')
 const multiaddr = require('multiaddr')
-const pull = require('pull-stream')
+const pipe = require('it-pipe')
+const { collect } = require('streaming-iterables')
 
-const Stardust = require('libp2p-stardust')
+const addr = multiaddr('/ip4/188.166.203.82/tcp/5892/ws/p2p-stardust')
+
+const stardust = new Stardust({ upgrader })
 
-Id.create((err, id) => { // generate a random id for testing
-  if (err) { throw err } // re-throw any error that might have occured
-
-  const peerInfo = new Info(id)
-  peerInfo.multiaddrs.add(multiaddr('/dns4/<TODO>/tcp/443/wss/p2p-stardust/'))
-
-  const stardust = new Stardust({ id }) // the id is required to prove the client's identity to the server
-
-  const modules = {
-    transport: [
-      stardust
-    ],
-    discovery: [
-      stardust.discovery
-    ]
-  }
-
-  const node = new Libp2p(modules, peerInfo) // create a libp2p node with the stardust transport
-
-  node.handle('/test/1.0.0', (protocol, conn) => {
-    pull(
-      pull.values(['hello']),
-      conn,
-      pull.map((s) => s.toString()),
-      pull.log()
-    )
-  })
-
-  node.start((err) => {
-    if (err) {
-      throw err
-    }
-
-    node.dial(peerInfo, '/test/1.0.0', (err, conn) => {
-      if (err) {
-        throw err
-      }
-
-      pull(
-        pull.values(['hello from the other side']),
-        conn,
-        pull.map((s) => s.toString()),
-        pull.log()
-      )
-    })
-  })
+const listener = stardust.createListener((socket) => {
+  console.log('new connection opened')
+  pipe(
+    ['hello'],
+    socket
+  )
 })
-```
 
-Outputs:
-```
-hello
-hello from the other side
-```
+await listener.listen(addr)
+console.log('listening')
 
-### Install
+const socket = await stardust.dial(addr)
+const values = await pipe(
+  socket,
+  collect
+)
 
-```bash
-> npm install libp2p-stardust
+console.log(`Value: ${values.toString()}`)
```

Comment:
values is an array, so I think that won't work

Also this example doesn't fully work, it would be great if the example in the readme would fully work on it's own

Reply:
I agree! I was thinking on getting it updated once we have a server deployed, so that we can point to it. 

I will  change the example so that it is "workable" and once we deploy it, we can put the correct server address :) I will create an issue for tracking this!